debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java

```diff
@@ -765,7 +765,7 @@ public boolean validateLogPosition(Partition partition, OffsetContext offset, Co
 
             LOGGER.info("Oldest SCN in logs is '{}'", oldestScn);
             LOGGER.info("Stored LSN is '{}'", storedLsn);
-            return storedLsn == null || Lsn.valueOf(oldestScn).compareTo(storedLsn) < 0;
+            return storedLsn.equals(Lsn.NULL) || Lsn.valueOf(oldestScn).compareTo(storedLsn) < 0;
```

Comment:
to avoid possible NPE, I'd use 
```suggestion
            return Lsn.NULL.equals(storedLsn) || Lsn.valueOf(oldestScn).compareTo(storedLsn) < 0;
```
but shouldn't we still consider the case when `storedLsn` is `null` in evaluating validity of the log position?

Reply:
Yeah, I totally agree! I was thinking the same but I wasn't so sure to be so _defensive_ (and I was a bit tired of the failing tests :eyes: ). If you think it's ok, I can evaluate also it

Reply:
Given a quick look seems that `storedLsn` should never be `null` but maybe it's better to also check for nullability